### PR TITLE
Ignore disabled radio buttons when moving selection.

### LINKED ISSUE #3839: RadioButton still selectable by keyboard when disabled
It appears that when a `RadioButton` is disabled, click events no longer respond as expected, but you can still navigate to it with the keyboard and select it with `space` or `enter`.

radio-select-disabled

```python
from textual.app import App, ComposeResult
from textual.widgets import RadioButton, RadioSet, Static


class Test(App):
    DEFAULT_CSS = """
    RadioButton:disabled {
        color: red;
    }
    """

    def compose(self) -> ComposeResult:
        with RadioSet():
            yield RadioButton("Enabled 1")
            yield RadioButton("Disabled 2", disabled=True)
            yield RadioButton("Disabled 3", disabled=True)
        yield Static("Selected:", id="selected")

    def on_radio_set_changed(self, radio_set: RadioSet.Changed) -> None:
        self.query_one("#selected").update(f"Selected: {radio_set.pressed.label}")


if __name__ == "__main__":
    app = Test()
    app.run()
```

# Textual Diagnostics

## Versions

| Name    | Value  |
|---------|--------|
| Textual | 0.44.1 |
| Rich    | 13.7.0 |

## Python

| Name           | Value                                                     |
|----------------|-----------------------------------------------------------|
| Version        | 3.8.10                                                    |
| Implementation | CPython                                                   |
| Compiler       | Clang 12.0.5 (clang-1205.0.22.9)                          |
| Executable     | /Users/geoff/Documents/projects/test/venv/bin/python |

## Operating System

| Name    | Value                                                                                                 |
|---------|-------------------------------------------------------------------------------------------------------|
| System  | Darwin                                                                                                |
| Release | 22.5.0                                                                                                |
| Version | Darwin Kernel Version 22.5.0: Thu Jun  8 22:22:19 PDT 2023; root:xnu-8796.121.3~7/RELEASE_ARM64_T8103 |

## Terminal

| Name                 | Value           |
|----------------------|-----------------|
| Terminal Application | vscode (1.85.0) |
| TERM                 | xterm-256color  |
| COLORTERM            | truecolor       |
| FORCE_COLOR          | *Not set*       |
| NO_COLOR             | *Not set*       |

## Rich Console options

| Name           | Value                |
|----------------|----------------------|
| size           | width=158, height=40 |
| legacy_windows | False                |
| min_width      | 1                    |
| max_width      | 158                  |
| is_terminal    | True                 |
| encoding       | utf-8                |
| max_height     | 40                   |
| justify        | None                 |
| overflow       | None                 |
| no_wrap        | False                |
| highlight      | None                 |
| markup         | None                 |
| height         | None                 |

### PR DESCRIPTION
**Pull request by rodrigogiraoserrao**
Fixes #3839

### CODE REVIEW

File: src/textual/widgets/_radio_set.py [hunk 1]
```diff
@@ -248,24 +248,59 @@ def action_previous_button(self) -> None:
...
+                Args:
+                    index: The index of the button to consider.
+
+                Returns:
+                    The distance between the two buttons.
+                """
+                return direction * (index - selected) % len(self.children)
+
+            with suppress(IndexError):
+                self._selected = sorted(candidate_indices, key=distance)[0]
```
Comment (willmcgugan) [inline src/textual/widgets/_radio_set.py:303]:
```suggestion
            self._selected = min(candidate_indices, key=distance, default=None)            
```

**Review by willmcgugan — APPROVED** [review]
Didn't quite follow your algorithm, but tests look good. Just one suggestion.

**Comment by rodrigogiraoserrao** [comment]
> Didn't quite follow your algorithm, but tests look good. Just one suggestion.

I'm happy to change the algorithm if you prefer.
I was just trying to avoid slices.

**Comment by willmcgugan** [comment]
It's totally fine if it works!

**Pull request merged**